replication-api-impl/src/test/java/org/codice/ditto/replication/api/impl/data/AbstractPersistableTest.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.impl.data;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.util.Map;
+import java.util.UUID;
+import org.junit.Test;
+
+public class AbstractPersistableTest {
+
+  @Test
+  public void testAbstractPersistable() {
+    String id = "testId";
+    AbstractPersistable persistable = new TestPersistable();
+    persistable.setId(id);
+    assertThat(persistable.getId(), equalTo(id));
+  }
+
+  @Test
+  public void generateId() {
+    AbstractPersistable persistable = new TestPersistable();
+    persistable.generateId();
+    UUID.fromString(persistable.getId()); // will throw an exception if a valid id wasn't generated

```

Comment:
❓ This test would fail for `ReplicatorConfig`s since we strip the `-` characters, does that mean we are saving that ID incorrectly? Should we mention this format in the API?

Reply:
@clockard do you know why we remove the `-` from our ids? Do we still need to do that?

Reply:
We had to do that when saving them as metacards since that is the way the catalog framework does it. I don't think there is a need to do it now.